node/index.js

```diff
@@ -582,6 +582,12 @@ TChannelConnection.prototype.onReaderError = function onReaderError(err) {
         localName: self.channel.hostPort,
         error: err
     });
+
+    // TODO instead of resetting send an error frame first.
+    // and reset the socket after sending an error frame
+    self.resetAll(new Error('tchannel read error'));
```

Comment:
I'll make it a wrapped error.

There might be value in having a seperate log call site for the tchannel read error instead of inferring it from the tchannel socket reset error where `err.type === 'tchannel.read-error'`